### PR TITLE
Testing 2

### PR DESCRIPTION
**Pull request by Thunae**
Add Junit and Mockito tests for servlet

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/ConvertLocationServlet.java [hunk 1]
```diff
@@ -30,6 +36,15 @@
 
 @WebServlet("/convert")
 public class ConvertLocationServlet extends HttpServlet {
+    private UrlOpener urlOpener;
+
+    public ConvertLocationServlet(){
```
Comment (lyze) [inline src/main/java/com/google/sps/servlets/ConvertLocationServlet.java]: nit: add space before parens

File: src/test/java/com/google/sps/ConvertLocationServletTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  @Mock
+  HttpServletResponse response;
+  @Mock
+  HttpSession session;
+  @Mock
+  UrlOpener urlOpener;
+  
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
```
Comment (lyze) [inline src/test/java/com/google/sps/ConvertLocationServletTest.java]: you don't need this call to initMocks if you use MockitoJUnitRunner

File: src/test/java/com/google/sps/LoginServletTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  @Mock
+  GoogleIdTokenVerifier verifier;
+  @Mock
+  GoogleIdToken token;
+  @Mock
+  Payload payload;
+  
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
```
Comment (lyze) [inline src/test/java/com/google/sps/LoginServletTest.java]: remove

File: src/test/java/com/google/sps/QueryServletTest.java [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+  @Mock
+  HttpSession session;
+  @Mock
+  UrlOpener urlOpener;
+  @Mock
+  RequestDispatcher rd;
+  
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
```
Comment (lyze) [inline src/test/java/com/google/sps/QueryServletTest.java]: remove

File: src/test/java/com/google/sps/SearchServletTest.java [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+      when(response.getWriter()).thenReturn(pw);
+
+      //Run the servlet and verify functions are called
+      new SearchServlet().doGet(request, response);
+      verify(response).setContentType("application/json;");
+      verify(response).getWriter();
+
+      //Process data and check correctness
+      String results = sw.getBuffer().toString().trim();
+      JsonObject resultsObj = (new JsonParser().parse(results)).getAsJsonArray().get(0).getAsJsonObject();
```
Comment (lyze) [inline src/test/java/com/google/sps/SearchServletTest.java]: remove these parens

**Review by lyze — COMMENTED** [review]
could you fix the commit message and author for the first commit in this PR?

Please also run the [java code formatter](https://github.com/google/google-java-format).

File: src/test/java/com/google/sps/ConvertLocationServletTest.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (lcatherine) [inline src/test/java/com/google/sps/ConvertLocationServletTest.java]: no wildcard imports

File: src/test/java/com/google/sps/QueryServletTest.java [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (lcatherine) [inline src/test/java/com/google/sps/QueryServletTest.java]: same

File: src/test/java/com/google/sps/SearchServletTest.java [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (lcatherine) [inline src/test/java/com/google/sps/SearchServletTest.java]: same

Comment (Thunae) [reply src/test/java/com/google/sps/SearchServletTest.java]: done

File: src/test/java/com/google/sps/QueryServletTest.java (hunk 7 above)
Comment (Thunae) [reply src/test/java/com/google/sps/QueryServletTest.java]: done

File: src/test/java/com/google/sps/ConvertLocationServletTest.java (hunk 6 above)
Comment (Thunae) [reply src/test/java/com/google/sps/ConvertLocationServletTest.java]: done

File: src/test/java/com/google/sps/SearchServletTest.java (hunk 5 above)
Comment (Thunae) [reply src/test/java/com/google/sps/SearchServletTest.java]: done

File: src/test/java/com/google/sps/QueryServletTest.java (hunk 4 above)
Comment (Thunae) [reply src/test/java/com/google/sps/QueryServletTest.java]: done

File: src/test/java/com/google/sps/LoginServletTest.java (hunk 3 above)
Comment (Thunae) [reply src/test/java/com/google/sps/LoginServletTest.java]: done

File: src/test/java/com/google/sps/ConvertLocationServletTest.java (hunk 2 above)
Comment (Thunae) [reply src/test/java/com/google/sps/ConvertLocationServletTest.java]: done

File: src/main/java/com/google/sps/servlets/ConvertLocationServlet.java (hunk 1 above)
Comment (Thunae) [reply src/main/java/com/google/sps/servlets/ConvertLocationServlet.java]: done

File: src/test/java/com/google/sps/ConvertLocationServletTest.java [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.junit.Assert.assertEquals;
```
Comment (lcatherine) [inline src/test/java/com/google/sps/ConvertLocationServletTest.java]: static statements come first before non static ones

File: src/test/java/com/google/sps/ConvertLocationServletTest.java [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+  HttpServletRequest request;
+  @Mock
+  HttpServletResponse response;
+  @Mock
+  HttpSession session;
+  @Mock
+  UrlOpener urlOpener;
+
+  @Test
+  public void GETZeroResults() throws IOException{
```
Comment (lcatherine) [inline src/test/java/com/google/sps/ConvertLocationServletTest.java]: camelCase

File: src/test/java/com/google/sps/ConvertLocationServletTest.java (hunk 9 above)
Comment (Thunae) [reply src/test/java/com/google/sps/ConvertLocationServletTest.java]: done

File: src/test/java/com/google/sps/ConvertLocationServletTest.java (hunk 10 above)
Comment (Thunae) [reply src/test/java/com/google/sps/ConvertLocationServletTest.java]: done

**Comment by lcatherine** [comment]
To change the commit message, use interactive rebase

`git rebase -i HEAD~2` (adjust the 2 according to number of commit messages you need to go back)

The interactive window will have 

`pick <commit number> <commit comment>
pick <commit number> <commit comment>
pick <commit number> <commit comment>`

Change `pick` to `reword` for the one you want to change the commit comment on.

To change the author, change `pick` to `edit` 
`git commit --amend --author="Author Name <email@address.com>"`

See [https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit]
[https://linuxize.com/post/change-git-commit-message/]

**Pull request closed without merging**